src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/HPackDecoder.cs

```diff
@@ -371,15 +513,21 @@ private void CheckIncompleteHeaderBlock(bool endHeaders)
             }
         }
 
-        private void ProcessHeaderValue(IHttpHeadersHandler? handler)
+        private void ProcessHeaderValue(ReadOnlySpan<byte> data, IHttpHeadersHandler? handler)
         {
-            OnString(nextState: State.Ready);
+            var headerNameSpan = _headerNameRange.Equals(default)
+                ? new Span<byte>(_headerName, 0, _headerNameLength)
+                : data[_headerNameRange];
```

Comment:
FWIW, I don't know how much these range slices matter, but the code quality for slicing a span with a Range isn't as good as slicing it with two ints, as the C# compiler generates code to extract the index and offset and do the math, and getting the index needs to check for negative values.  If these slicing code paths are super hot, you might want to consider using something lower than Range.  As an example, this code `data[_headerNameRange]` effectively gets expanded into the equivalent of:
```C#
int tmpLength = data.Length;
Range r = _headerNameRange;
int start = r.Start.GetOffset(length);
int length = r.End.GetOffset(length) - start;
return data.Slice(start, length);
```
and you can see an approximate impact on the asm here:
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNYgGYmpBkJoBvGgzMMxUXguwYYDNti4BzewH0AFjGwATGQDlsAUcXGA1qcwsrGzsGXh4GT28/KECBWQxsKAxwyMtrW3t4jESvXwCgmAAZGBcMD1zzU3MdB2SAeS5JAE9ZMScAHmBuuwA+BllJZRhyAAo2Dq7e/q4hkZhxn1tsAEoGAF5R5sizLcyAbSTy1MqQ1wBdRrNjpl0F306evsHhsYmplSkeaLL4rNZ/M67A5HCInU7bFiTaazK4pNIwDJZDBoUrJCoCGp1Dw7cIAXyAA=

That said, if this code isn't super hot, the readability/maintainability of using Ranges is more important.

Reply:
I might change them to `(int start, int length)?`. The "from end" functionality of Range isn't required, and `data[_headerNameRange]` vs `data.Slice(_headerNameRange.start, _headerNameRange.length)` isn't a big deal.